dataclients/kubernetes/kubernetestest/api.go

```diff
@@ -43,8 +44,9 @@ type api struct {
 func NewAPI(o TestAPIOptions, specs ...io.Reader) (*api, error) {
 	a := &api{
 		namespaces: make(map[string]namespace),
+		// see https://kubernetes.io/docs/reference/using-api/api-concepts/#resource-uris
 		pathRx: regexp.MustCompile(
-			"(/namespaces/([^/]+))?/(services|ingresses|routegroups|endpointslices|endpoints|secrets)",
+			"(?:/namespaces/([^/]+))?/(services|ingresses|routegroups|endpointslices|endpoints|secrets)(?:/(.+))?",
```

Comment:
We may consider refactoring this to use https://pkg.go.dev/net/http#hdr-Patterns added in go1.22

Reply:
What do you think about having test like this? For you not to forget to have a look when we will update Go version.
```
func TestPathRx(t *testing.T) {
    if runtime.Version() == "go1.22" {
        t.Error("please pay attention to dataclients/kubernetes/kubernetestest/api.go:49 using http.Patterns")
    }
}
```


https://pkg.go.dev/runtime#Version

Reply:
This will fail as we use 1.22 already.
We can use 1.22 features after we bump min version in [go.mod](https://github.com/zalando/skipper/blob/ca9abef2b36b1e650365efe2d4b8e8dac69dae59/go.mod#L184)

The proposed change is optional it may wait until we touch this code again.